netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,14 +79,18 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
```

Comment:
Ah I missed the comment, thanks!  I'm still wondering if the comment is enough though.  I suspect if it's an `InetSocketAddress`, we would always want to use `forTarget`.  Otherwise, I suspect users are going to continually shoot themselves in the foot.